packages/superset-ui-control-utils/src/types.ts

```diff
@@ -228,34 +233,42 @@ export type SharedSectionAlias =
   | 'sqlaTimeSeries'
   | 'NVD3TimeSeries';
 
-export interface ControlItem {
+export interface OverrideSharedControlItem {
   name: SharedControlAlias;
-  config: Partial<ControlConfig>;
+  override: Partial<SharedControlConfig>;
 }
 
-export interface CustomControlItem {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface CustomControlItem<P = any> {
   name: string;
-  config: ControlConfig;
+  config: CustomComponentControlConfig<P>;
 }
 
-export type ControlSetItem = SharedControlAlias | ControlItem | CustomControlItem | ReactNode;
+export type ControlSetItem =
+  | SharedControlAlias
+  | OverrideSharedControlItem
+  | CustomControlItem
+  | ReactElement
```

Comment:
would be nice to add explanation why 
`ReactNode` vs `ReactElement | null`

Reply:
`ReactNode` also includes `ReactText` which contains `string`. I wanted `string` to be exclusive for `SharedControlAlias`, so just by `if (typeof controlItem !== 'string')`, TypeScript will be able to infer what the remaining types `controlItem` could be.

Reply:
ic.

Reply:
perhaps can add comment in code. in case someone stop by and wanting to change it to `ReactNode` without knowing the context

Reply:
good call, will do